### DISCUSSION TITLE
CI Tiers and Cadences

### DISCUSSION THREAD

**Discussion by Bckempa** [space-ros/space-ros#128] [Ideas]
As the number of lints and test grows and as more static analysis is added to the workflow CI times are going to balloon.

Since there has been discussion about focusing on static analysis for the April release we should consider breaking up the build and test flows to better accommodate build times 

For example:
  - PRs get a lint and smoke-test build on push
  - Full test-suite ran daily
  - Full static analysis (since it requires significant developer time to parse and disposition the results, many alerts will be upstream and not under direct project control, and overall should be slow-moving) bi-weekly in advance of the technical meeting

**Comment by EzraBrooks** [comment]
I think step 1 is optimizing build times down and doing more simple bring-up tests in CI - the multi-hour runtimes of the moveit2 build, for example, should be less frequent.
